src/VisualStudio/Core/Def/Implementation/Workspace/VisualStudioActiveDocumentTracker.cs

```diff
@@ -60,12 +52,12 @@ internal class VisualStudioActiveDocumentTracker : ForegroundThreadAffinitizedOb
             : base(threadingContext, assertIsForeground: false)
         {
             _editorAdaptersFactoryService = editorAdaptersFactoryService;
-            _asyncTasks = new JoinableTaskCollection(threadingContext.JoinableTaskContext);
-            _asyncTasks.Add(ThreadingContext.JoinableTaskFactory.RunAsync(async () =>
+            ThreadingContext.JoinableTaskCollection.Add(ThreadingContext.JoinableTaskFactory.RunAsync(async () =>
```

Comment:
I'm not sure if renaming IThreadingContext.JoinableTaskCollection to something a bit less vague will help clarity here -- if we call it something roughly like "DeferredTaskCollection" it's a bit more obvious that's what's going on here?

Reply:
`DisposalToken` and `JoinableTaskCollection` match the corresponding properties in `AsyncPackage`. We still have an open question about whether it's worth the overhead to apply this to `JoinableTaskFactory` as well (likely we will, but as a follow-up).

Reply:
If you were going to rename it, I'd call it `ShutdownBlockingTasks` or something meaningful like that.